trace/config.go

```diff
@@ -269,6 +281,25 @@ func WithStackTrace(b bool) SpanEndEventOption {
 	return stackTraceOption(b)
 }
 
+type errorStatusOption bool
+
+func (o errorStatusOption) applyEvent(c EventConfig) EventConfig {
+	c.errorStatus = bool(o)
+	return c
+}
+
+func (o errorStatusOption) applySpanEnd(c SpanConfig) SpanConfig {
+	c.errorStatus = bool(o)
+	return c
+}
+
+var _ SpanEndEventOption = errorStatusOption(true)
+
+// WithErrorStatus sets the flag to set span's status to error if error or panic is occurred.
+func WithErrorStatus(b bool) SpanEndEventOption {
```

Comment:
Do we need the boolean? If we call the option, we want it enabled. Otherwise, we just don't add the parameter?

Reply:
Hm, yes, you are right.
I wanted to make it consistent with other options like `WithStackTrace`
https://github.com/open-telemetry/opentelemetry-go/blob/fb7cc020f24aae0fcb42f2e11c1db78bb0a297fc/trace/config.go#L268

However, I am not sure if it's acceptable to use options like:
`span.RecordError(err, trace.WithStackTrace(true), trace.WithErrorStatus())`

What do you think?

Reply:
I tend to think options with a boolean don't need that parameter. But for consistency, we could maybe still have it here.
Let's wait for other opinions.

Reply:
Additionally I thought about splitting `WithErrorStatus` option:
- `WithStatus` for `span.RecordError` call.
- `WithStatusOnPanic` for `span.End` call.
Because it may be not clear what means `WithErrorStatus` option when you call `span.End`.

But that also creates inconsistency with `WithStackTrace` option 🙃

Reply:
I'm thinking the `WithStatusOnPanic` name would be nice, as it makes things clearer.